packages/expressive-code/src/index.ts

```diff
@@ -1,16 +1,21 @@
-import { ExpressiveCodeEngine, ExpressiveCodeEngineConfig, ExpressiveCodePlugin } from '@expressive-code/core'
+import type { ExpressiveCodePlugin } from '@expressive-code/core'
+import { type ExpressiveCodeCoreConfig, ExpressiveCodeCore } from './core'
 import type { PluginFramesOptions } from '@expressive-code/plugin-frames'
 import { pluginFrames } from '@expressive-code/plugin-frames'
 import type { PluginShikiOptions } from '@expressive-code/plugin-shiki'
 import { pluginShiki } from '@expressive-code/plugin-shiki'
 import { pluginTextMarkers } from '@expressive-code/plugin-text-markers'
 
+// NOTE - tsup code splitting is disabled to avoid re-exports being dropped
+// see https://github.com/evanw/esbuild/issues/1737 & https://github.com/evanw/esbuild/issues/1521
+// TODO: Consider a different bundler in order to support code-splitting to reduce /dist size (e.g., unbuild)
```

Comment:
I had to disable splitting in `expressive-code` due to limitations with `tsup/esbuild` and `re-exports`.  I noticed that you also have it disabled in [@expressive-code/core](https://github.com/expressive-code/expressive-code/blob/main/packages/%40expressive-code/core/package.json#L38), likely for the same reason.

This isn't ideal but also not the end of the world since the total size of `dist` doesn't increase all that much but possibly it may be worth considering a bundler that has better support for `re-exports` such as [unbuild](https://github.com/unjs/unbuild) which would, more or less, be a drop-in replacement for tsup.